### PR TITLE
tf integration fixes

### PR DESCRIPTION
**Pull request by jgrayson676**
## Description of the change

Adds error handling to the claim integration and fixes an incorrectly typed field in the data service integration.

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change
- [ ] Technical Debt
- [ ] Documentation

## Related tickets

https://app.clubhouse.io/active-prospect/story/20093/verify-error-handling-in-tf-integration
https://app.clubhouse.io/active-prospect/story/21131/incorrect-response-variable-data-type

### Development and Testing

- [ ]  Lint rules pass locally.
- [ ]  The code changed/added as part of this pull request has been covered with tests, or this PR does not alter production code.
- [ ]  All tests related to the changed code pass in development, or tests are not applicable.

### Code Review

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ]  At least two engineers have been added as "Reviewers" on the pull request.
- [ ]  Changes have been reviewed by at least two other engineers who did not write the code.
- [ ]  This branch has been rebased off master to be current.

### Tracking 
- [ ]  Issue from Clubhouse has a link to this pull request.
- [ ]  This PR has a link to the issue in Clubhouse.

### QA
- [ ]  This branch has been deployed to staging and tested.

### CODE REVIEW

File: lib/claim.js [hunk 1]
```diff
@@ -28,10 +24,14 @@ const handle = (vars, callback) => {
   };
 
   client.claim(options, (err, res, body) => {
-
+    if (err && !body) {
```
Comment (cgrayson) [inline lib/claim.js:27]: Is there a case where both `err` and `body` are defined?

**Review by cgrayson — APPROVED** [review]
Otherwise LGTM!

File: lib/claim.js (hunk 1 above)
Comment (jgrayson676) [reply lib/claim.js:27]: yeah
https://github.com/activeprospect/trustedform-api-client/blob/e569ea2cf0d65f8beef09d89cd046d108b1919a7/index.js#L46

I didn't remember why it was written that way and half-wondered if that should be changed but checking for the existence of `body` worked in this case.

Comment (cgrayson) [reply lib/claim.js:27]: How about something like this? Always handle presence of `err`, but get the better `reason` value from `parseResponse()` (which seemed pretty safe to me to call even in an error state):

```js
  client.claim(options, (err, res, body) => {
    if (err) {
      const parsed = parseResponse(res, body, vars);
      return callback(null, {
        outcome: 'error',
        reason: parsed.reason || err.message
      });
    }
```

**Review by cgrayson — APPROVED** [review]
LGTM

**Review by cgrayson — APPROVED** [review]

**Pull request merged**